sklearn/metrics/tests/test_score_objects.py

```diff
@@ -1140,3 +1145,34 @@ def test_scorer_no_op_multiclass_select_proba():
         labels=lr.classes_,
     )
     scorer(lr, X_test, y_test)
+
+
+@pytest.mark.parametrize("name, scorer", SCORERS.items())
+def test_scorer_metadata_request(name, scorer):
+    assert hasattr(scorer, "set_score_request")
+    assert hasattr(scorer, "get_metadata_routing")
+
+    weighted_scorer = scorer.set_score_request(sample_weight=True)
+    # set_score_request shouldn't mutate the instance
+    assert weighted_scorer is not scorer
+
+    assert str(scorer.get_metadata_routing()) == "{}"
+    assert (
+        str(weighted_scorer.get_metadata_routing())
+        == "{'score': {'sample_weight': <RequestType.REQUESTED: True>}}"
+    )
+
+    # make sure putting the scorer in a router doesn't request anything
+    router = MetadataRouter(owner="test").add(scorer=scorer, method_mapping="score")
```

Comment:
```suggestion
    router = MetadataRouter(owner="test").add(method_mapping="score", scorer=scorer)
```
Not sure if this is better or worse, maybe worse.

Reply:
I really don't mind either way 😆